### PR TITLE
Set CCSID and Delivery mode

### PR DESCRIPTION
**Pull request by AnumehaSrivastava05**
Set CCSID to 819
Set Delivery mode to NON_PERSISTENT

### CODE REVIEW

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateJMSMessageHeader.java [hunk 1]
```diff
@@ -14,13 +14,17 @@
  * Add hexadecimal message ControlId with a prefix ID: to add provider specific 
  * value to stop any encoding in MQRFH2.
  * Set custom JMS message Id by adding MQMD.
+ * Set CCSID to 819
+ * Set DeliveryMode to 'MQPER_NOT_PERSISTENT
```
Comment (weskubo-cgi) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateJMSMessageHeader.java:18]: I assume MQPER_NOT_PERSISTENT is code 1 but might want to make that clear in the comment. Also there's only a opening single quote and no closing quote.

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/BaseRoute.java [hunk 2]
```diff
@@ -16,8 +16,7 @@
 public abstract class BaseRoute extends RouteBuilder {
 	protected static final String JMS_DESTINATION_NAME_FORMAT = "queue:///%s?targetClient=1&&mdWriteEnabled=true";
 	
-	protected static final String MQ_URL_FORMAT = "jmsComponent:queue:%s?exchangePattern=InOut&replyTo=queue:///%s&replyToType=Shared&allowAdditionalHeaders=JMS_IBM_MQMD_MsgId";
-	
+	protected static final String MQ_URL_FORMAT = "jmsComponent:queue:%s?exchangePattern=InOut&replyTo=queue:///%s&replyToType=Exclusive&allowAdditionalHeaders=JMS_IBM_MQMD_MsgId&deliveryMode=1&requestTimeout=7000&explicitQosEnabled=true&timeToLive=7000";
```
Comment (daveb-hni) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/BaseRoute.java:19]: Both JMSDeliveryMode in the exchange header and deliveryMode in the url are set. Is this required? It mentions in the documentation that one of these takes priority based on the setting of the preserveMessageQos. I don't think the header will be considered as this property has a default of 'false'.

Comment (daveb-hni) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/BaseRoute.java:19]: replyToType has been overwritten from Shared to Exclusive.

Comment (daveb-hni) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/BaseRoute.java:19]: We probably need to start adding some comments about why we are using certain properties and their values here for properties like requestTimeout, timeToLive. Or a general about the expected endpoint behaviour. I can add comment on replyToType of 'Shared' being required as multiple apps could be connecting to the same queue and this indicates Camel needs to be selective about which messages it attempts to retrieve.

**Review by daveb-hni — CHANGES_REQUESTED** [review]

**Review by daveb-hni — APPROVED** [review]

**Pull request merged**